tests/test_featurizers/sparse_featurizer_checks.py

```diff
@@ -0,0 +1,71 @@
+from rasa.nlu.tokenizers.tokenizer import Tokenizer
+from rasa.shared.nlu.constants import TEXT
+from rasa.nlu.constants import TOKENS_NAMES
+from rasa.shared.nlu.training_data.message import Message
+import itertools as it
+
+
+def test_component_requires_tokenizer(tokenizer, featurizer, msg):
+    """The component should have a tokenizer in its required components."""
+    req_components = featurizer.__class__.required_components()
+    predicate = [c == Tokenizer for c in req_components]
+    assert any(predicate)
+
+
+def test_component_no_features_on_no_tokens(tokenizer, featurizer, msg):
+    """The component does not set any sparse features if there are no tokens."""
+    message = Message({TEXT: msg})
+    featurizer.process(message)
+    seq_vecs, sen_vecs = message.get_sparse_features(TEXT, [])
+    assert not seq_vecs
+    assert not sen_vecs
```

Comment:
I'm actually not 100% sure if this is expected behavior. I could be wrong, but shouldn't we set an empty vector here? Otherwise DIET may break, no?